### PR TITLE
Revamped Design

### PR DESCRIPTION
**Pull request by 5A11**
This PR updates the docs website's design. Specifically:
* Removes faulty edit page function
* Updates google analytics integration (old method is deprecating)
* Enables light and dark themes (following system's preference and user's choice)
* Correctly applies fetch favicon
* Enables fast page loading
* Enables automatic URL tracking of active anchors
* Adds back-to-top button
* Adds search keyword autocompletion
* Adds search occurrence highlighting
* Adds search and result deep link sharing
* Enables "announcements" and adds $150M developer fund as the current announcement
* Adds explanations to MkDoc's various configuration options
* Enables modern code block feature
* Enables annotations feature
* Enables modern content tab feature
* Enables modern tables feature
* Adds social links to footer
* Updates footer design
* Adds copyright notice to footer
* Updates many dependencies to their latest versions
* Removes extra assets and files
* Updates the index page to be in line with the rest of the material design
* Removes some extra (unused) docs pages
* Adds makefile and updates readme accordingly

### CODE REVIEW

**Review by daeMOn63 — DISMISSED** [review]

File: docs/index.md [hunk 1]
```diff
@@ -1,40 +1,28 @@
...
+
+### What is Fetch.ai
+
+There are billions of devices, processes and pipelines around the world that are desperate for automation and communication. Most blockchains aren't built to handle such a large number of transactions, and none enable their clients to learn from the data generated and exchanged.
+
+Fetch.ai is an ecosystem built to bring value to data, by connecting devices and facilitating the creation of new markets based on what they learn from each other. It is where truly autonomous decisions can be made.
+
+### Joining our networks
+
+Learn how to use our [ledger technologies](../ledger_v2) to join our mainnet v2
```
Comment (jrriehl) [inline docs/index.md]:
```suggestion
Learn how to use our [ledger technologies](../ledger_v2) to join our mainnet v2.
```

Comment (jrriehl) [reply docs/index.md]: Dark theme link color could be a bit brighter, otherwise looks good to me.

Comment (5A11) [reply docs/index.md]: That's the colour pallet that comes by default on dark mode. Can change the colours manually but I would prefer somebody with design experience to set it. Can be done later in another PR.

**Comment by GataKamsky** [comment]
Looks good. Developer Fund should now get more attention.

**Review by GataKamsky — APPROVED** [review]
Looks good. Developer Fund should now receive more attention.

**Pull request merged**